### PR TITLE
[NEW] Added omnichannel sorting disclaimer to SortModeList menu

### PR DESCRIPTION
**Pull request by aleksandernsilva**
## Proposed changes (including videos or screenshots)

Screen Shot 2023-01-31 at 15 32 07

## Further comments

OC-568

### CODE REVIEW

**Review by filipemarins — APPROVED** [review]

File: apps/meteor/client/components/Omnichannel/OmnichannelSortingDisclaimer.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+export const OmnichannelSortingDisclaimer = () => {
+	const t = useTranslation();
+	const sortingMechanism = useSetting('Omnichannel_sorting_mechanism') as OmniSortingType;
+
+	const [{ [sortingMechanism]: type }] = useState({
+		[OmniSortingType.Priority]: t('Priorities'),
+		[OmniSortingType.SLAs]: t('SLA_Policies'),
+		[OmniSortingType.Timestamp]: '',
+	});
```
Comment (ggazzo) [inline apps/meteor/client/components/Omnichannel/OmnichannelSortingDisclaimer.tsx:14]: there is no need to translate all the keys just to pick one later

File: apps/meteor/client/components/Omnichannel/OmnichannelSortingDisclaimer.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+	if (!type) {
+		return null;
+	}
+
+	return (
+		<Box is='li' padding='4px 12px' fontSize='14px' lineHeight='20px' maxWidth='180px'>
+			{t('Omnichannel_sorting_disclaimer', { sortingMechanism: type })}
+		</Box>
```
Comment (ggazzo) [inline apps/meteor/client/components/Omnichannel/OmnichannelSortingDisclaimer.tsx:23]: I'm sure this should not be a `LI` and this item should be the `aria-describedby` for the sorting option

File: apps/meteor/client/components/SortList/SortModeList.tsx [hunk 3]
```diff
@@ -30,6 +31,8 @@ function SortModeList(): ReactElement {
 					text={t('Name')}
 					input={<RadioButton pis='x24' name='sidebarSortby' onChange={setToAlphabetical} checked={sidebarSortBy === 'alphabetical'} />}
 				/>
+
+				<OmnichannelSortingDisclaimer />
```
Comment (ggazzo) [inline apps/meteor/client/components/SortList/SortModeList.tsx:35]: why not just render this IF you are a agent/manager and the omnichannel is enabled?

**Review by ggazzo — CHANGES_REQUESTED** [review]
please check the comments

File: apps/meteor/client/components/Omnichannel/OmnichannelSortingDisclaimer.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { OmnichannelSortingMechanismSettingType as OmniSortingType } from '@rocket.chat/core-typings';
+import { Box } from '@rocket.chat/fuselage';
+import { useSetting, useTranslation } from '@rocket.chat/ui-contexts';
+import React, { useState } from 'react';
+
+export const OmnichannelSortingDisclaimer = () => {
+	const t = useTranslation();
+	const sortingMechanism = useSetting('Omnichannel_sorting_mechanism') as OmniSortingType;
```
Comment (ggazzo) [inline apps/meteor/client/components/Omnichannel/OmnichannelSortingDisclaimer.tsx:8]:
```suggestion
	const sortingMechanism = useSetting<OmniSortingType>('Omnichannel_sorting_mechanism');
```

File: apps/meteor/client/components/Omnichannel/OmnichannelSortingDisclaimer.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+		[OmniSortingType.Timestamp]: '',
+	} as const);
+
+	if (!type) {
+		return null;
+	}
+
+	return (
+		<Box is='small' display='block' padding='4px 12px' fontSize='14px' lineHeight='20px' maxWidth='180px' {...props}>
+			{t('Omnichannel_sorting_disclaimer', { sortingMechanism: t(type) })}
```
Comment (KevLehman) [inline apps/meteor/client/components/Omnichannel/OmnichannelSortingDisclaimer.tsx:26]: Isn't this an EE only disclaimer? Priorities don't exist on CE

**Pull request merged**